os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -544,27 +515,34 @@ static int reloading_thread(int argc, char *argv[])
 		bmdbg("Terminate binary %d\n", bidx);
 	}
 
-	/* 2. Unload common binary */
+	/* Unload common binary */
 	ret = binary_manager_terminate_binary(BM_CMNLIB_IDX);
 	if (ret != OK) {
 		bmdbg("Failed to terminate common binary\n");
 		return BINMGR_OPERATION_FAIL;
 	}
+#else
+	load_cmd = LOADCMD_LOAD;
 
-	/* 3. Load all binaries */
-	ret = binary_manager_execute_loader(LOADCMD_LOAD_ALL, 0);
+	/* Unload the faulty binary */
+	ret = binary_manager_terminate_binary(bin_idx);
 	if (ret != OK) {
-		bmdbg("Failed to execute loader to load all binaries\n");
+		bmdbg("Failed to terminate binary %s\n", BIN_NAME(bin_idx));
 		return BINMGR_OPERATION_FAIL;
 	}
+#endif
 
-	return BINMGR_OK;
-#else
-	/* argv[1] binary index for reloading */
-	int bin_idx = (int)atoi(argv[1]);
+	/* Deinitialize modules in kernel */
+	binary_manager_deinit_modules();
 
-	return binary_manager_reload(bin_idx);
-#endif
+	/* Create a loader to reload binary */
+	ret = binary_manager_execute_loader(load_cmd, bin_idx);
```

Comment:
Most of the binary manager functions return `BINMGR_XXX`.
How about changing the return of `binary_manager_execute_loader` and the return check logic?

Reply:
I will modify all later.